addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -136,482 +171,192 @@ static func _category_cmp(a: BlockCategory, b: BlockCategory) -> bool:
 	return a.name.naturalcasecmp_to(b.name) < 0
 
 
-static func get_categories(blocks: Array[Block], extra_categories: Array[BlockCategory] = []) -> Array[BlockCategory]:
+static func get_categories(blocks: Array[BlockDefinition], extra_categories: Array[BlockCategory] = []) -> Array[BlockCategory]:
 	var cat_map: Dictionary = {}
 	var extra_cat_map: Dictionary = {}
 
 	for cat in extra_categories:
 		extra_cat_map[cat.name] = cat
 
 	for block in blocks:
-		var cat: BlockCategory = cat_map.get(block.category)
+		var block_cat_name: String = block.category
```

Comment:
Can you name this `block_category_name`? Or maybe just `name` would work in this context?